src/AutoRest.CSharp/Mgmt/Decorator/ScopeDetection.cs

```diff
@@ -58,10 +59,20 @@ private static RequestPath CalculateScopePath(RequestPath requestPath)
         {
             var indexOfProvider = requestPath.ToList().LastIndexOf(Segment.Providers);
             // if there is no providers segment, myself should be a scope request path. Just return myself
-            if (indexOfProvider < 0)
-                return requestPath;
-
-            return new RequestPath(requestPath.Take(indexOfProvider));
+            if (indexOfProvider >= 0)
+            {
+                if (indexOfProvider == 0 && requestPath.SerializedPath.StartsWith("/providers/Microsoft.Management/managementGroups", StringComparison.InvariantCultureIgnoreCase))
+                    return RequestPath.ManagementGroup;
+                return new RequestPath(requestPath.Take(indexOfProvider));
+            }
+            // Returns RequestPath.ResourceGroup for the list path "/subscriptions/{subscriptionId}/resourceGroups" as well.
+            if (requestPath.SerializedPath.StartsWith("/subscriptions/{subscriptionId}/resourceGroups", StringComparison.InvariantCultureIgnoreCase))
+                return RequestPath.ResourceGroup;
+            if (requestPath.SerializedPath.StartsWith("/subscriptions", StringComparison.InvariantCultureIgnoreCase))
+                return RequestPath.Subscription;
+            if (requestPath.SerializedPath.Equals("/tenants"))
+                return RequestPath.Tenant;
+            return requestPath;
```

Comment:
I do not like hard-coded strings here, which might lead to problems in the future.
Looks like we are testing if these paths are subscription, resource group, mgmt group. How about we get the resource type from these paths, and switch the resource type?

Reply:
This mainly takes care of the cases of the list path for scopes. For instance, previously `/subscriptions` returns itself as scope, then it does not match the scope of `/subscriptions/{subscriptionId}`,  now it returns `/subscriptions/{subscriptionId}` so the scopes can match. `/subscriptions` is not a valid ResourceIdentifier, I'm not sure how to get its resource type.